src/packages/client/src/runtime/utils/find.ts

```diff
@@ -0,0 +1,290 @@
+ * @returns
+ */
+function isMatched(string: string, regexs: (RegExp | string)[]) {
+  for (const regex of regexs) {
+    if (typeof regex === 'string') {
+      if (string.includes(regex)) {
+        return true
+      }
+    } else if (regex.exec(string)) {
+      return true
+    }
+  }
+
+  return false
+}
+
+/**
+ * Find paths that match a set of regexes
+ * @param root to start from
+ * @param match to match against
+ * @param types to select files, folders, links
+ * @param deep to recurse in the directory tree
+ * @param limit to limit the results
+ * @param handler to further filter results
+ * @param found to add to already found #private
+ * @param seen to add to already seen #private
+ * @returns unresolved paths
+ */
+export function findSync(
+  root: string,
+  match: (RegExp | string)[],
+  types: ('f' | 'd' | 'l')[] = ['f', 'd', 'l'],
+  deep: ('d' | 'l')[] = [],
+  limit: number = Infinity,
+  handler: Handler = () => true,
+  found: string[] = [],
+  seen: Record<string, true> = {},
+) {
+  try {
+    const realRoot = realpathSync(root)
+
+    // we make sure not to loop infinitely
+    if (seen[realRoot]) {
+      return found
+    }
+
+    // we stop if we found enough results
+    if (limit - found.length <= 0) {
+      return found
+    }
+
+    // we check that the root is a directory
+    if (direntToType(statSync(realRoot)) !== 'd') {
+      return found
+    }
+
+    // we list the items in the current root
+    const items = readdirSync(root, { withFileTypes: true })
+
+    //seen[realRoot] = true
+    for (const item of items) {
+      // we get the file info for each item
+      const itemName = item.name
+      const itemType = direntToType(item)
+      const itemPath = path.join(root, item.name)
+
+      // if the item is one of the selected
+      if (itemType && types.includes(itemType)) {
+        // if the path of an item has matched
+        if (isMatched(itemPath, match)) {
+          const value = handler(root, itemName, itemType)
+
+          // if we changed the path value
+          if (typeof value === 'string') {
+            found.push(value)
+          }
+          // if we kept the default path
+          else if (value === true) {
+            found.push(itemPath)
+          }
+        }
+      }
+
+      if (deep.includes(itemType as any)) {
+        // dive within the directory tree
+        // we recurse and continue mutating `found`
+        findSync(itemPath, match, types, deep, limit, handler, found, seen)
+      }
+    }
+  } catch {}
+
+  return found
+}
+
+/**
+ * Like `findSync` but moves to the parent folder if nothing is found
+ * @param root to start from
+ * @param match to match against
+ * @param types to select files, folders, links
+ * @param deep to recurse in the directory tree
+ * @param limit to limit the results
+ * @param filter to further filter results
+ * @param found to add to already found #private
+ * @param seen to add to already seen #private
```

Comment:
What does #private means?

Reply:
It's a field used for recursion mostly, I could remove that as it does not convey much meaning

Reply:
Btw that file is duplicated and that is normal, it's related to another issue where we are not able to generate types for cases where we import from another local package. For example, it would fail if `sdk` would export `find` and `find` were imported in `client`. This only fails on functions and is related to `rollup-plugin-dts`. Will open an issue soon.

Reply:
Is it not possible to import like this from client? That could work I think.
`import something from '@prisma/sdk/dist/find`

Reply:
Yeah we already use `import something from '@prisma/sdk/dist/xxx` in the runtime